### PR TITLE
feat(combobox): Added group type to items

### PR DESCRIPTION
**Pull request by yaminideveloper**
# Description

Please include a summary of the change and which issue is fixed. Please also
include relevant motivation and context. List any dependencies that are required
for this change.

# Associated Tickets

- SPRK-

# Checklist:

- [ ] I've updated unit tests where needed
- [ ] I've updated the components README.md where needed
- [ ] I've added major version bumps for my breaking changes
- [ ] I've added changesets where needed
- [ ] All the components I've updated are reflected in the changelog

### CODE REVIEW

**Comment by lukebennett88** [comment]
Hi @yaminideveloper, thanks for the PR.
We use Changesets to manage versioning. You could you please add one 🙏
https://github.com/brighte-labs/spark-web/blob/main/CONTRIBUTING.md#version-management

**Comment by nderkim** [comment]
@lukebennett88 is this something we want to add? If so, it requires some more work to override the group styles to match our designs. I don't think we have designs for groups yet either.

File: packages/combobox/src/combobox.tsx [hunk 1]
```diff
@@ -29,7 +29,7 @@ export type ComboboxProps<Item = unknown> = {
   inputValue?: string;
 
   /** Array of items for the user to select from. */
-  items: Awaitable<Item[]>;
+  items: Awaitable<(Item | GroupBase<Item>)[]>;
```
Comment (clothoo) [inline packages/combobox/src/combobox.tsx:32]: Don't we need to pass it to react-select?

Comment (nderkim) [reply packages/combobox/src/combobox.tsx:32]: Not necessarily. Our API intentionally only exposes a subset of the many features supported by `react-select`. If we want to add grouped items to our list of supported features, we will also need to add our styles to support it.

**Comment by lukebennett88** [comment]
Updated styles for Group headings:
PixelSnap 2022-05-26 at 16 59 13@2x

**Review by lukebennett88 — APPROVED** [review]

**Pull request merged**